src/pivot-table/hooks/use-column-width/use-column-width-left.ts

```diff
@@ -45,47 +77,35 @@ export default function useColumnWidthLeft({ layoutService, tableRect, headersDa
-              // Use the max width of all measures
-              cellWidth = maxMeasureCellWidth;
+          let columnWidth: ColumnWidth | undefined;
+          if (lastRowHeader.columnWidthLocation === ColumnWidthLocation.Pivot) {
+            columnWidth = layoutService.layout.qHyperCube.topHeadersColumnWidth;
+          } else if (lastRowHeader.columnWidthLocation === ColumnWidthLocation.Dimension) {
+            ({ columnWidth } = lastRowHeader);
+          }
+
+          let fitToContentWidth = 0;
+          if (lastRowHeader.isLeftDimension) {
+            if (lastRowHeader.id === PSEUDO_DIMENSION_KEY) {
+              fitToContentWidth = maxMeasureColumnWidth();
             } else {
-              const { label, qApprMaxGlyphCount, columnWidth, isLocked } = header;
-              const expandIconSize =
-                !isFullyExpanded && header.isLeftDimension && collIdx < qNoOfLeftDims - 1 ? EXPAND_ICON_SIZE : 0;
-              const lockedIconSize = isLocked ? LOCK_ICON_SIZE : 0;
-              const measuredHeaderWidth =
-                measureTextForHeader(label) + TOTAL_CELL_PADDING + MENU_ICON_SIZE + lockedIconSize;
-
-              let fitToContentWidth = 0;
-              if (header.isLeftDimension) {
-                fitToContentWidth = Math.max(
-                  measuredHeaderWidth,
-                  estimateWidthForDimensionValue(qApprMaxGlyphCount as number) + TOTAL_CELL_PADDING + expandIconSize,
-                );
-              } else if (lastRowLastColumn && !header.isLeftDimension && hasPseudoDimOnLeft) {
-                fitToContentWidth = maxMeasureCellWidth;
-              } else {
-                fitToContentWidth = measuredHeaderWidth;
-              }
-
-              cellWidth = getColumnWidthValue(tableRect.width, columnWidth, fitToContentWidth);
+              fitToContentWidth = Math.max(
+                dimensionsFitToContentWidth(collIdx),
+                dimensionValuesFitToContentWidth(lastRowHeader, collIdx),
+              );
             }
-
-            // The last cell setting should override the other cells in that column, so we don't pick the max
-            const isTypeAuto = !header.columnWidth || header.columnWidth.type === ColumnWidthType.Auto;
-            if (lastRowLastColumn && !isTypeAuto) {
-              return cellWidth;
+          } else if (!lastRowHeader.isLeftDimension) {
+            fitToContentWidth = dimensionsFitToContentWidth(collIdx);
+            if (lastRowHeader.columnWidthLocation === ColumnWidthLocation.Measures) {
+              fitToContentWidth = maxMeasureColumnWidth(fitToContentWidth);
             }
+          }
 
-            return Math.max(maxWidth, cellWidth);
-          }, width);
+          if (columnWidth && columnWidth.type !== ColumnWidthType.Auto) {
+            width = getColumnWidthValue(tableRect.width, columnWidth, fitToContentWidth);
+          } else {
+            width = fitToContentWidth;
+          }
```

Comment:
I think you don't need the else case. getColumnWidthValue should basically have the same logic, when columnWidth is undefined or type is auto, it will use return fitToContentWidth

Reply:
You are correct